syntax = "proto3";  
package banyan; 
option go_package = "banyan/pa";

message HardClause {
    repeated int32 lits = 2;
}

message SoftClause {
    int32 weight = 1;
    repeated int32 lits = 2;
}

message Problem {
    int32 clusterId = 1;
    int32 numVars = 2;
    int32 numClauses = 3;
    int32 topWeight = 4;
    repeated HardClause hardClauses = 5;
    repeated SoftClause softClauses = 6;
}

message Problems {
    repeated Problem problems = 1;
}

message Result {
    int32 status = 1;
    repeated int32 lits = 2;
}

message Results {
    map<int32,Result> clusterRst = 1;
}